File: facebook-login.js
const puppeteer = require('puppeteer');
require('dotenv').config();
(async () => {
  const url = 'https://facebook.com';
  const user = process.env.USER;
  const password = process.env.PASSWORD;
  const browserOptions = {
    headless: false,
    defaultViewport: null,
    args: ['--start-maximized'],
  };
  const browser = await puppeteer.launch(browserOptions);
  const page = await browser.newPage();
  await page.goto(url);
  await page.type('#email', user);
  await page.type('#pass', password);
  await page.$eval('#login_form', (form) => form.submit());
  await page.screenshot({ path: 'example.png' });
  // await browser.close();
})();

File: facebook_login.py
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium import webdriver
import os

if __name__ == "__main__":
    url = 'https://facebook.com'
    gecko_path = os.getenv('FIREFOX_PATH', '.\geckodriver.exe')
    browser = webdriver.Firefox(executable_path=gecko_path)
    email = os.getenv('EMAIL', 'user@example.com')
    password = os.getenv('PASSWORD', 'password')
    browser.get(url)
    browser.find_element_by_id('email').send_keys(email)
    browser.find_element_by_id('pass').send_keys(password)
    browser.save_screenshot('example.png')
    # Sometimes using geckodriver the layout changes
    try:
        browser.find_element_by_id('login_form').submit()
    except:
        browser.find_element_by_id('u_0_a').submit()
    # self.browser.quit()
